pandas/core/algorithms.py

```diff
@@ -1534,40 +1538,52 @@ def _take_nd_object(arr, indexer, out, axis: int, fill_value, mask_info):
 }
 
 
+@functools.lru_cache(maxsize=128)
+def __get_take_nd_function_cached(ndim, arr_dtype, out_dtype, axis):
+    """
+    Part of _get_take_nd_function below that doesn't need the mask
+    and thus can be cached.
+    """
+    tup = (arr_dtype.name, out_dtype.name)
+    if ndim == 1:
+        func = _take_1d_dict.get(tup, None)
+    elif ndim == 2:
+        if axis == 0:
+            func = _take_2d_axis0_dict.get(tup, None)
```

Comment:
The content of this function is simply copied out from `_get_take_nd_function` below, no other changes to the code have been made

Reply:
just checking, this is _cut_ from the existing function, not _copied_?  i.e. this isnt introducing a big chunk of duplicate code?

Reply:
Yes, cut (I just meant to say that no logic changed inside it)